cmd/cmd.go

```diff
@@ -904,6 +931,25 @@ func generateInteractive(cmd *cobra.Command, opts generateOptions) error {
 
 		if len(prompt) > 0 && multiline == MultilineNone {
 			opts.Prompt = prompt
+			if multiModal {
+				newPrompt, images, err := extractFileNames(prompt)
+				if err != nil {
+					return err
+				}
+				opts.Prompt = newPrompt
+
+				// reset the context if we find another image
+				if len(images) > 0 {
+					opts.Images = images
+					ctx := cmd.Context()
+					ctx = context.WithValue(ctx, generateContextKey("context"), []int{})
+					cmd.SetContext(ctx)
+				}
+				if len(opts.Images) == 0 {
```

Comment:
I don't know if this is a good idea. At its core, a llava model is still a language model and it's possible to interact with it as just a text completion or chat model

Reply:
Was thinking this too! It works very well sans-image

Reply:
I'm extremely reluctant to take this out because with the llava model it's pretty much useless until an image is added. I get that you can get it to answer a question, but that feels like a degenerate usecase for the current model. The kicker for me is that there is no indication to the user that they can even add an image.

Reply:
LLaVA is better at understanding the context of text and images and using that information to answer questions or generate text than Llama2.

if my sample.txt file is:

C'est l'histoire d'un petit chat qui chantait "miaou miaou" tout le temps.

And translate it also in Italian

if my prompt is :

ollama run llama2 --verbose please translate in spanish "$(cat /Users/igor/sample.txt)"

The answer will be:

Spanish: "Es la historia de un pequeño gato que cantaba 'miau miau' todo el tiempo."

Italian: "Questa è la storia di un gattino che cantava 'miao miao' sempre."

The text in the sample.txt file is sent to Ollama and is interpreted as being part of the prompt, not part of the file that must be processed.

By separating the text from the prompt, LLaVA can focus on processing the text content independently, leading to more efficient and accurate responses. This approach could be particularly useful for tasks that require extensive text processing, such as translation or summarization.